Mathlib/Data/Prod/Basic.lean

```diff
@@ -300,21 +300,43 @@ theorem map_involutive [Nonempty α] [Nonempty β] {f : α → α} {g : β → 
     Involutive (map f g) ↔ Involutive f ∧ Involutive g :=
   map_leftInverse
 
-section delaborators
+namespace PrettyPrinting
 open Lean PrettyPrinter Delaborator
 
+/--
+When true, then `Prod.fst x` and `Prod.snd x` pretty print as `x.1` and `x.2`
+rather than as `x.fst` and `x.snd`.
+-/
+register_option pp.numericProj.prod : Bool := {
+  defValue := true
+  descr := "enable pretty printing `Prod.fst x` as `x.1` and `Prod.snd x` as `x.2`."
+}
+
+/-- Tell whether pretty-printing should use numeric projection notations `.1`
+and `.2` for `Prod.fst` and `Prod.snd`. -/
+def getPPNumericProjProd (o : Options) : Bool :=
+  o.get pp.numericProj.prod.name pp.numericProj.prod.defValue
+
 /-- Delaborator for `Prod.fst x` as `x.1`. -/
 @[app_delab Prod.fst]
-def delabProdFst : Delab := withOverApp 3 do
-  let x ← SubExpr.withAppArg delab
-  `($(x).1)
+def delabProdFst : Delab :=
+  whenPPOption getPPNumericProjProd <|
+  whenPPOption getPPFieldNotation <|
+  whenNotPPOption getPPExplicit <|
```

Comment:
Give this is the option users are most likely to change, should it be checked first for speed?

Reply:
I expect it would be very difficult to measure any performance impact.